include/exadg/compressible_navier_stokes/postprocessor/output_generator.cpp

```diff
@@ -162,69 +166,43 @@ void
-      // In the first time step, the current time might be larger than start_time. In that
-      // case, we first have to reset the counter in order to avoid that output is written every
-      // time step.
-      if(reset_counter)
-      {
-        if(time > output_data.start_time)
-        {
-          output_counter +=
-            int((time - output_data.start_time + EPSILON) / output_data.interval_time);
-        }
-        reset_counter = false;
-      }
-
-
-      if(time > (output_data.start_time + output_counter * output_data.interval_time - EPSILON))
-      {
-        pcout << std::endl
-              << "OUTPUT << Write data at time t = " << std::scientific << std::setprecision(4)
-              << time << std::endl;
-
-        write_output<dim, Number, VectorType>(output_data,
-                                              *dof_handler,
-                                              *mapping,
-                                              solution_conserved,
-                                              additional_fields,
-                                              output_counter,
-                                              mpi_comm);
-
-        ++output_counter;
-      }
-    }
-    else
-    {
-      pcout << std::endl
-            << "OUTPUT << Write " << (output_counter == 0 ? "initial" : "solution") << " data"
-            << std::endl;
-
-      write_output<dim, Number, VectorType>(output_data,
-                                            *dof_handler,
-                                            *mapping,
-                                            solution_conserved,
-                                            additional_fields,
-                                            output_counter,
-                                            mpi_comm);
-
-      ++output_counter;
-    }
+    pcout << std::endl
+          << "OUTPUT << Write data at time t = " << std::scientific << std::setprecision(4) << time
+          << std::endl;
+
+    write_output<dim, Number, VectorType>(output_data,
+                                          *dof_handler,
+                                          *mapping,
+                                          solution_conserved,
+                                          additional_fields,
+                                          time_control.get_counter(),
+                                          mpi_comm);
+  }
+  else
+  {
+    pcout << std::endl
+          << "OUTPUT << Write " << (time_control.get_counter() == 0 ? "initial" : "solution")
+          << " data" << std::endl;
+
+    write_output<dim, Number, VectorType>(output_data,
+                                          *dof_handler,
+                                          *mapping,
+                                          solution_conserved,
+                                          additional_fields,
+                                          time_control.get_counter(),
+                                          mpi_comm);
```

Comment:
The function call seems to be the same as above for the unsteady case. So we should pull this out of the if-else-branch and call the function only once.